File: OOP in Java - Homework/Problem 1. Geometry/src/Geometry/Shapes/PlaneShapes/Circle.java
package Geometry.Shapes.PlaneShapes;

import Geometry.Shapes.PlaneShape;
import Geometry.Vertices.Vertex2D;

public class Circle extends PlaneShape {
    private double radius;

    public Circle(Vertex2D centerPoint, double radius) throws Exception {
        this.setRadius(radius);
        this.vertices.add(centerPoint);
    }

    public void setRadius(double radius) throws Exception {
        if(radius < 0){
            throw new Exception("Radius cannot be negative number");
        }
        this.radius = radius;
    }

    public double getRadius(){
        return this.radius;
    }

    @Override
    public double getArea() {
        return Math.PI * Math.pow(this.getRadius(), 2);
    }

    @Override
    public double getPerimeter() {
        return 2 * Math.PI * this.getRadius();
    }

    @Override
    public String toString(){
        return (new StringBuilder())
                .append(super.toString())
                .append("\nType - Circle")
                .append("\nRadius: " + this.getRadius())
                .toString();
    }
}

File: Encapsulation and Polymorphism - Homework/Problem 1. Shapes/Circle.cs
namespace Problem_1.Shapes
{
    using System;

    public class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public Circle()
            : this(0)
        {

        }
        public double Radius
        {
            get { return this.radius; }
            set
            {
                if(value < 0)
                {
                    throw new ArgumentException("The radius cannot be negative.");
                }
                else
                {
                    this.radius = value;
                }
            }
        }

        public double CalculateArea()
        {
            return Math.PI * Math.Pow(this.Radius, 2);
        }

        public double CalculatePerimeter()
        {
            return 2 * Math.PI * this.Radius;
        }

        public override string ToString()
        {
            return "I am a circle";
        }
    }
}
